cortex-js/src/infrastructure/commanders/models/model-pull.command.ts

```diff
@@ -1,25 +1,151 @@
+import { RepoDesignation, listFiles } from '@huggingface/hub';
+import YAML from 'yaml';
+import * as readline from 'node:readline/promises';
+import { basename } from 'node:path';
 
 @SubCommand({
   name: 'pull',
   aliases: ['download'],
   description: 'Download a model. Working with HuggingFace model id.',
 })
 export class ModelPullCommand extends CommandRunner {
+  private metadataFileName = 'metadata.yaml';
+
   constructor(private readonly modelsCliUsecases: ModelsCliUsecases) {
     super();
   }
 
   async run(input: string[]) {
     if (input.length < 1) {
-      console.error('Model ID is required');
+      console.error('Model Id is required');
       exit(1);
     }
 
-    await this.modelsCliUsecases.pullModel(input[0]);
+    // Check if metadata.yaml file exist
+    const metadata = await this.getJanMetadata(input[0]);
+
+    if (!metadata) {
+      await this.modelsCliUsecases.pullModel(input[0]);
+    } else {
+      await this.handleJanHqModel(input[0], metadata);
+    }
+
     console.log('\nDownload complete!');
     exit(0);
   }
+
+  private async getJanMetadata(input: string): Promise<any> {
+    // try to append with janhq/ if it's not already
+    const sanitizedInput = input.trim().startsWith('janhq/')
+      ? input
+      : `janhq/${input}`;
+
+    const repo: RepoDesignation = { type: 'model', name: sanitizedInput };
+    let isMetadataFileExist = false;
+    for await (const fileInfo of listFiles({ repo })) {
+      if (fileInfo.path === this.metadataFileName) {
+        isMetadataFileExist = true;
+        break;
+      }
+    }
+
+    if (!isMetadataFileExist) {
+      return undefined;
+    }
+
+    const path = `https://huggingface.co/${sanitizedInput}/raw/main/${this.metadataFileName}`;
+    const res = await fetch(path);
+    const metadataJson = await res.text();
+    const parsedMetadata = YAML.parse(metadataJson);
+    return parsedMetadata;
+  }
+
+  private async versionInquiry(tags: string[]): Promise<string> {
+    return new Promise((resolve) => {
+      let selectedTag = 'default';
+      let prompt = 'Select the version you want to download:\n';
+      for (let i = 0; i < tags.length; i++) {
+        prompt += `${i}. ${tags[i]}\n`;
+      }
+      prompt += '>> ';
+
+      const rl = readline.createInterface({
+        input: stdin,
+        output: stdout,
+        prompt: prompt,
+      });
+      rl.prompt();
+
+      rl.on('close', () => {
+        resolve(selectedTag);
+      });
+
+      rl.on('line', (input) => {
+        if (input.trim().length === 0) {
+          rl.close();
+        }
+
+        try {
+          if (Number(input) >= 0 && Number(input) < tags.length) {
+            selectedTag = tags[Number(input)];
+            rl.close();
+          } else {
+            console.error('Invalid version');
+            rl.prompt();
+          }
+        } catch (e) {
+          console.error('Invalid version');
+          rl.prompt();
+        }
+      });
+    });
+  }
```

Comment:
Why not use Choices? It's a lot shorter

Reply:
ah yes, I forgot about the `inquirerService`